XenAdmin/TabPages/GeneralTabPage.cs

```diff
@@ -1943,6 +1943,11 @@ private void buttonViewLog_Click(object sender, EventArgs e)
             StartPutty( "env docker logs --tail=50 --follow --timestamps");
         }
 
+        private void panel2_Paint(object sender, PaintEventArgs e)
+        {
+            panel2.AutoScroll = sections.Sum(s => s.Height) > panel2.Height;

```

Comment:
Each of the sections sits in a panel that has a top and bottom padding, so I think we need to sum the parents' height here rather than the sections'. Also, the sum should probably be compared with the height of panel2's client area which excludes borders, scrollbars etc. (even if we don't have horizontal scrollbar here).